trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveDataTypeTransformer.java

```diff
@@ -6,28 +6,52 @@
 package com.datastrato.gravitino.trino.connector.catalog.hive;
 
 import com.datastrato.gravitino.rel.types.Type;
-import com.datastrato.gravitino.rel.types.Type.Name;
 import com.datastrato.gravitino.rel.types.Types;
+import com.datastrato.gravitino.trino.connector.GravitinoErrorCode;
 import com.datastrato.gravitino.trino.connector.util.GeneralDataTypeTransformer;
+import io.trino.spi.TrinoException;
+import io.trino.spi.type.VarcharType;
 
 /** Type transformer between Hive and Trino */
 public class HiveDataTypeTransformer extends GeneralDataTypeTransformer {
   // Max length of Hive varchar is 65535
   private static final int HIVE_VARCHAR_MAX_LENGTH = 65535;
+  private static final int HIVE_CHAR_MAX_LENGTH = 255;
 
   @Override
   public Type getGravitinoType(io.trino.spi.type.Type type) {
-    Type gravitinoType = super.getGravitinoType(type);
-    if (gravitinoType.name() == Name.VARCHAR
-        && ((Types.VarCharType) gravitinoType).length() > HIVE_VARCHAR_MAX_LENGTH) {
-      return Types.VarCharType.of(HIVE_VARCHAR_MAX_LENGTH);
-    }
+    Class<? extends io.trino.spi.type.Type> typeClass = type.getClass();
+    if (typeClass == VarcharType.class) {
+      VarcharType varcharType = (VarcharType) type;
+      if (varcharType.getLength().isEmpty()) {
+        // It was creating a table with column type 'varchar' NOT 'varchar(n)', We do not support
+        // this case
```

Comment:
Why does Hive not support the datatype varchar without specifying a length?  It has conflict with https://trino.io/docs/current/connector/hive.html#basic-usage-examples

Reply:
Data type varchar without length corresponds to `String` type in Gravitino and there is NO data type `String` actually. 

Please see: 
https://docs.google.com/document/d/1xNkl4Y2t8Bfb8UOOdGBhyQr5DDreAOLmK8FOWiIDiQM/edit